starknet/starknet-normalization/src/mapping.ts

```diff
@@ -0,0 +1,95 @@
+import type * as rpc from '@subsquid/starknet-data'
+import {Block, BlockHeader, Transaction, Event, TransactionType, PriceUnit} from './data'
+import {addErrorContext} from '@subsquid/util-internal'
+
+export function mapRpcBlock(src: rpc.Block): Block {
+    let header: BlockHeader = {
+        hash: src.block_hash,
+        height: src.block_number,
+        parentHash: src.parent_hash,
+        status: src.status,
+        newRoot: src.new_root,
+        timestamp: src.timestamp,
+        sequencerAddress: src.sequencer_address
+    }
+
+    let events: Event[] = []
+
+    let transactions = src.transactions
+        ?.map((tx, i) => {
+            try {
+                return mapRpcTransaction(i, tx, events)
+            } catch(err: any) {
+                throw addErrorContext(err, {
+                    blockTransaction: tx.transaction.transaction_hash
+                })
+            }
+        }) ?? []
+
+    return {
+        header,
+        transactions,
+        events
+    }
+}
+
+
+function mapRpcTransaction(
+    transactionIndex: number,
+    src: rpc.PackedTransaction,
+    events: Event[]
+): Transaction {
+    const tx = src.transaction;
+    const receipt = src.receipt;
+    
+    // Create transaction object with all possible fields
+    const transaction: Transaction = {
+        transactionIndex,
+        transactionHash: tx.transaction_hash,
+        contractAddress: tx.contract_address || undefined,
+        entryPointSelector: tx.entry_point_selector || undefined,
+        calldata: tx.calldata || undefined,
```

Comment:
can `calldata` or `constructorCalldata` be an empty string? if so then result will be `undefined` but it's better to keep the empty string. also it would interpret 0 tip as `undefined`. here my suggestion to substitute every `|| undefined` with `?? undefined`